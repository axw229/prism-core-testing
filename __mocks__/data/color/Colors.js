// @flow
import cloneDeep from 'lodash/cloneDeep'
import sample from 'lodash/sample'

import type { ColorSetPayload, Color } from '../../../src/shared/types/Colors'

export const Colors: ColorSetPayload = {
  "Red": [
    [
      {
        "colorNumber": "6561",
        "coordinatingColors": {
          "coord2ColorId": 1926,
          "coord1ColorId": 2250,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2248",
        "isExterior": true,
        "isInterior": true,
        "name": "Teaberry",
        "rgb": 15454683,
        "lrv": 68.551,
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 235,
        "green": 209,
        "blue": 219,
        "hue": 0.935897435897436,
        "saturation": 0.39393939393939387,
        "lightness": 0.8705882352941177,
        "hex": "#ebd1db",
        "isDark": false,
        "cssrgb": "rgb(235,209,219)",
        "storeStripLocator": "102-C1",
        "similarColors": [
          "2255",
          "2262",
          "2269",
          "2662",
          "2535",
          "2276",
          "2658",
          "1974",
          "2543",
          "2531"
        ]
      },
      {
        "colorNumber": "6568",
        "coordinatingColors": {
          "coord2ColorId": 11244,
          "coord1ColorId": 2687,
          "whiteColorId": 2681
        },
        "description": [
          "Subdued",
          "Muted",
          "Brilliant",
          "Radiant"
        ],
        "id": "2255",
        "isExterior": true,
        "isInterior": true,
        "name": "Lighthearted Pink",
        "rgb": 15586781,
        "lrv": 70.758,
        "brandedCollectionIds": [
          31797,
          31791
        ],
        "brandedCollectionNames": [
          "Pottery Barn Fall/Winter 2018",
          "Pottery Barn Fall/Winter 2018"
        ],
        "colorFamilyNames": [
          "Red"
        ],
        "brandKey": "SW",
        "red": 237,
        "green": 213,
        "blue": 221,
        "hue": 0.9444444444444445,
        "saturation": 0.3999999999999999,
        "lightness": 0.8823529411764706,
        "hex": "#edd5dd",
        "isDark": false,
        "cssrgb": "rgb(237,213,221)",
        "storeStripLocator": "103-C1",
        "similarColors": [
          "2248",
          "2262",
          "2269",
          "2276",
          "1974",
          "1981",
          "2658",
          "1995",
          "2662",
          "1988"
        ]
      }
    ]
  ]
}

export function getColors (): ColorSetPayload {
  return cloneDeep(Colors)
}

export function getColor (): Color | void {
  const colors = getColors()
  const fam1 = Object.keys(colors)[0]
  if (colors[fam1]) {
    return sample(colors[fam1][0])
  }
}
