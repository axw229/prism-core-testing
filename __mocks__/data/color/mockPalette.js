const mockPalette = {
  'colors': [
    {
      'colorNumber': '6765',
      'coordinatingColors': {
        'coord2ColorId': '11194',
        'coord1ColorId': '2870',
        'whiteColorId': '2162'
      },
      'description': [
        'Fairly Colorful',
        'Moderately Colorful',
        'Pale'
      ],
      'id': '2452',
      'isExterior': true,
      'isInterior': true,
      'name': 'Spa',
      'lrv': 64.476,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Blue'
      ],
      'brandKey': 'SW',
      'red': 167,
      'green': 220,
      'blue': 220,
      'hue': 0.5,
      'saturation': 0.43089430894308944,
      'lightness': 0.7588235294117647,
      'hex': '#a7dcdc',
      'isDark': false,
      'storeStripLocator': '164-C2',
      'similarColors': [
        '2634',
        '2630',
        '11225',
        '2472',
        '2445',
        '2622',
        '2459',
        '1606',
        '11226',
        '2451'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '6860',
      'coordinatingColors': {
        'coord2ColorId': '11339',
        'coord1ColorId': '11364',
        'whiteColorId': '2681'
      },
      'description': [
        'Soft',
        'Fairly Bright'
      ],
      'id': '2547',
      'isExterior': true,
      'isInterior': true,
      'name': 'Eros Pink',
      'lrv': 18.874,
      'brandedCollectionNames': [
        '2020 Play',
        '2019 Naturalist'
      ],
      'colorFamilyNames': [
        'Red'
      ],
      'brandKey': 'SW',
      'red': 200,
      'green': 79,
      'blue': 104,
      'hue': 0.9655647382920111,
      'saturation': 0.5238095238095238,
      'lightness': 0.5470588235294118,
      'hex': '#c84f68',
      'isDark': false,
      'storeStripLocator': '105-C5',
      'similarColors': [
        '2273',
        '2266',
        '2542',
        '2546',
        '1998',
        '2280',
        '2530',
        '2287',
        '2279',
        '30193'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9018',
      'coordinatingColors': {
        'coord2ColorId': '11218',
        'coord1ColorId': '11200',
        'whiteColorId': '2366'
      },
      'description': [
        'Intense',
        'Pale'
      ],
      'id': '11195',
      'isExterior': false,
      'isInterior': true,
      'name': 'Honey Bees',
      'lrv': 70.14,
      'brandedCollectionNames': [
        '2018 Unity'
      ],
      'colorFamilyNames': [
        'Yellow'
      ],
      'brandKey': 'SW',
      'red': 251,
      'green': 214,
      'blue': 130,
      'hue': 0.11570247933884299,
      'saturation': 0.937984496124031,
      'lightness': 0.7470588235294118,
      'hex': '#fbd682',
      'isDark': false,
      'storeStripLocator': '134-C4',
      'similarColors': [
        '2382',
        '11199',
        '11196',
        '2588',
        '2375',
        '11198',
        '11194',
        '11191',
        '3146',
        '2361'
      ],
      'ignore': false,
      'archived': false
    },
    {
      'colorNumber': '9007',
      'coordinatingColors': {
        'coord2ColorId': '1666',
        'coord1ColorId': '2686',
        'whiteColorId': '2331'
      },
      'description': [
        'Rich',
        'Vibrant',
        'Pale'
      ],
      'id': '11184',
      'isExterior': false,
      'isInterior': true,
      'name': 'Melón Meloso',
      'lrv': 55.314,
      'brandedCollectionNames': [],
      'colorFamilyNames': [
        'Orange'
      ],
      'brandKey': 'SW',
      'red': 242,
      'green': 184,
      'blue': 140,
      'hue': 0.0718954248366013,
      'saturation': 0.7968749999999999,
      'lightness': 0.7490196078431373,
      'hex': '#f2b88c',
      'isDark': false,
      'storeStripLocator': '121-C3',
      'similarColors': [
        '2334',
        '2983',
        '2327',
        '11188',
        '2326',
        '2045',
        '2038',
        '3012',
        '2341',
        '2031'
      ],
      'ignore': false,
      'archived': false
    }
  ],
  'activeColor': {
    'colorNumber': '9007',
    'coordinatingColors': {
      'coord2ColorId': '1666',
      'coord1ColorId': '2686',
      'whiteColorId': '2331'
    },
    'description': [
      'Rich',
      'Vibrant',
      'Pale'
    ],
    'id': '11184',
    'isExterior': false,
    'isInterior': true,
    'name': 'Melón Meloso',
    'lrv': 55.314,
    'brandedCollectionNames': [],
    'colorFamilyNames': [
      'Orange'
    ],
    'brandKey': 'SW',
    'red': 242,
    'green': 184,
    'blue': 140,
    'hue': 0.0718954248366013,
    'saturation': 0.7968749999999999,
    'lightness': 0.7490196078431373,
    'hex': '#f2b88c',
    'isDark': false,
    'storeStripLocator': '121-C3',
    'similarColors': [
      '2334',
      '2983',
      '2327',
      '11188',
      '2326',
      '2045',
      '2038',
      '3012',
      '2341',
      '2031'
    ],
    'ignore': false,
    'archived': false
  },
  'toggleCompareColor': false,
  'temporaryActiveColor': null
}

export default mockPalette
