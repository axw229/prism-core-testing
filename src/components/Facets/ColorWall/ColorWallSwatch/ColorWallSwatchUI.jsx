// @flow
import React, { useMemo, forwardRef, useEffect } from 'react'
import { Link } from 'react-router-dom'
import { CLASS_NAMES } from './shared'
import { type Color } from '../../../../shared/types/Colors'
import { arrayToSpacedString } from '../../../../shared/helpers/StringUtils'
import { fullColorName } from '../../../../shared/helpers/ColorUtils'

import './ColorWallSwatch.scss'
import 'src/scss/convenience/visually-hidden.scss'

type Props = {
  color: Color,
  focus?: boolean,
  onClick?: Function,
  tabIndex?: number,
  thisLink: string
}

const _classes = arrayToSpacedString([
  CLASS_NAMES.BASE,
  CLASS_NAMES.BASE_CLICKABLE
])

const ColorWallSwatchUI = forwardRef<Props, Object>((props: Props, ref: Object) => {
  const { focus, color, thisLink, onClick, tabIndex = 0 } = props

  useEffect(() => {
    ref.current = {
      link: thisLink,
      onClick: onClick
    }
  }, [thisLink, onClick])

  const handleClick = useMemo(() => (e: any) => {
    if (typeof onClick === 'function') {
      onClick(e)
    }
  }, [onClick])

  const linkProps = useMemo(() => ({
    to: thisLink,
    style: {
      background: color.hex
    }
  }), [thisLink, color])

  const linkClasses = useMemo(() => `${_classes} ${focus ? CLASS_NAMES.BASE_FOCUS : ''}`, [focus])
  const colorName = useMemo(() => fullColorName(color.brandKey, color.colorNumber, color.name), [color])

  return (
    <div className={CLASS_NAMES.SWATCH}>
      <Link {...linkProps}
        className={linkClasses}
        onClick={handleClick}
        tabIndex={tabIndex}>
        <span className='visually-hidden'>
          {colorName}
        </span>
      </Link>
    </div>
  )
})

export default ColorWallSwatchUI
