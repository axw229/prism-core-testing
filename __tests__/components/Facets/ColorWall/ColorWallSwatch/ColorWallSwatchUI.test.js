/* eslint-env jest */
import React from 'react'
import ColorWallSwatchUI from 'src/components/Facets/ColorWall/ColorWallSwatch/ColorWallSwatchUI'
import * as Colors from '__mocks__/data/color/Colors'
import { Link } from 'react-router-dom'
const color = Colors.getColor()

const clickFn = jest.fn()
const defaultProps = {
  color: color,
  thisLink: '',
  focus: true,
  ref: { current: null },
  onClick: clickFn
}

test('should rendring Link correctly with props', () => {
  expect(mocked(<ColorWallSwatchUI {...defaultProps} thisLink='link' />).find(Link).exists()).toEqual(true)
})

test('click function will be invoked when link is clicked', () => {
  mocked(<ColorWallSwatchUI {...defaultProps} thisLink='link' />).find(Link).simulate('click')
  expect(clickFn).toHaveBeenCalled()
})
