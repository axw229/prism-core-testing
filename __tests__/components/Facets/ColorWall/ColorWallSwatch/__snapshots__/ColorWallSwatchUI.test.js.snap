// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test ColorWallSwatchUI should match the Snapshot 1`] = `
<div
  className="color-wall-swatches__swatch"
>
  <Link
    className="color-swatch-inner color-swatch-inner--clickable color-swatch-inner--focus"
    onClick={[Function]}
    style={
      Object {
        "background": "#ebd1db",
      }
    }
    tabIndex={-1}
    to=""
  >
    <span
      className="visually-hidden"
    >
      SW 6561 Teaberry
    </span>
  </Link>
</div>
`;
